src/hapi/pipelines/database/operational_presence.py

```diff
@@ -82,63 +87,67 @@ def populate(self, debug=False):
+                        if not sector_code:
+                            add_missing_value_message(
+                                errors, dataset_name, "sector", sector_orig
+                            )
+                            continue
+
+                        # * Admin processing
                         if admin_level == "admintwo":
                             country_code = self._admintwo.pcode_to_iso3.get(
                                 admin_code
                             )
-                        if admin_level == "adminone":
+                        elif admin_level == "adminone":
                             country_code = self._adminone.pcode_to_iso3.get(
                                 admin_code
                             )
-                        org_info = self._org.get_org_info(
-                            org_name_orig, location=country_code
+                        else:
+                            country_code = admin_code
+                        admin2_code = admins.get_admin2_code_based_on_level(
+                            admin_code=admin_code, admin_level=admin_level
                         )
-                        org_name = org_info.get("#org+name")
-                        self._org.add_org_to_lookup(org_name_orig, org_name)
-                        org_acronym = org_info.get(
-                            "#org+acronym",
-                            values[org_acronym_index][admin_code][i],
+                        admin2_ref = self._admins.admin2_data[admin2_code]
+
+                        # * Org processing
+                        if not org_str:
+                            org_str = values[org_acronym_index][admin_code][i]
+                        (
+                            org_name,
+                            org_name_normalise,
+                            org_acronym,
+                            org_acronym_normalise,
+                            org_type_code,
+                        ) = self._org.get_org_info(
+                            org_str, location=country_code
                         )
-                        if org_acronym is not None and len(org_acronym) > 32:
-                            org_acronym = org_acronym[:32]
-                        org_type_code = org_info.get("#org+type+code")
-                        org_type_name = None
-                        if not org_type_code:
+
+                        if org_acronym is None:
+                            org_acronym = values[org_acronym_index][
+                                admin_code
+                            ][i]
+                            if org_acronym is not None:
+                                if len(org_acronym) > 32:
+                                    org_acronym = org_acronym[:32]
+
+                                org_acronym_normalise = normalise(org_acronym)
+
+                        # * Org type processing
+                        if org_type_code is None:
```

Comment:
If you combine these to a single line it will reduce the nesting

Reply:
Good idea!